core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -87,43 +88,43 @@ class SharedPreferenceUtil @Inject constructor(context: Context?) {
     sharedPreferences.getString(PREF_STORAGE_TITLE, defaultTitle)
 
   fun putPrefLanguage(language: String?) =
-    sharedPreferences.edit().putString(PREF_LANG, language).apply()
+    sharedPreferences.edit { putString(PREF_LANG, language) }
 
   fun putPrefIsFirstRun(isFirstRun: Boolean) =
-    sharedPreferences.edit().putBoolean(PREF_IS_FIRST_RUN, isFirstRun).apply()
+    sharedPreferences.edit { putBoolean(PREF_IS_FIRST_RUN, isFirstRun) }
 
   fun putPrefWifiOnly(wifiOnly: Boolean) =
-    sharedPreferences.edit().putBoolean(PREF_WIFI_ONLY, wifiOnly).apply()
+    sharedPreferences.edit { putBoolean(PREF_WIFI_ONLY, wifiOnly) }
 
   fun putPrefStorageTitle(storageTitle: String?) =
-    sharedPreferences.edit().putString(PREF_STORAGE_TITLE, storageTitle).apply()
+    sharedPreferences.edit { putString(PREF_STORAGE_TITLE, storageTitle) }
 
   fun putPrefStorage(storage: String) {
-    sharedPreferences.edit().putString(PREF_STORAGE, storage).apply()
+    sharedPreferences.edit { putString(PREF_STORAGE, storage) }
     prefStorages.onNext(storage)
   }
 
   fun putPrefFullScreen(fullScreen: Boolean) =
-    sharedPreferences.edit().putBoolean(PREF_FULLSCREEN, fullScreen).apply()
+    sharedPreferences.edit { putBoolean(PREF_FULLSCREEN, fullScreen) }
 
   fun putPrefExternalLinkPopup(externalLinkPopup: Boolean) =
-    sharedPreferences.edit().putBoolean(PREF_EXTERNAL_LINK_POPUP, externalLinkPopup).apply()
+    sharedPreferences.edit { putBoolean(PREF_EXTERNAL_LINK_POPUP, externalLinkPopup) }
 
   fun showIntro(): Boolean = sharedPreferences.getBoolean(PREF_SHOW_INTRO, true)
 
-  fun setIntroShown() = sharedPreferences.edit().putBoolean(PREF_SHOW_INTRO, false).apply()
+  fun setIntroShown() = sharedPreferences.edit { putBoolean(PREF_SHOW_INTRO, false) }
 
   var showHistoryAllBooks: Boolean
     get() = sharedPreferences.getBoolean(PREF_SHOW_HISTORY_ALL_BOOKS, true)
     set(prefShowHistoryAllBooks) {
-      sharedPreferences.edit().putBoolean(PREF_SHOW_HISTORY_ALL_BOOKS, prefShowHistoryAllBooks)
-        .apply()
+      sharedPreferences.edit { putBoolean(PREF_SHOW_HISTORY_ALL_BOOKS, prefShowHistoryAllBooks) }
```

Comment:
We can make this setter's constructor's functionality directly to its delegation like:

```set(prefShowHistoryAllBooks) = {sharedPreferences.edit {tBoolean(PREF_SHOW_HISTORY_ALL_BOOKS, prefShowHistoryAllBooks) }```

Should this be done?
And one more question, what is the exact name of the change that i just suggested via using terms like _delegation_, i used that just because of its literal meaning

Reply:
![image](https://user-images.githubusercontent.com/3358282/87923306-2fca6880-ca75-11ea-9ab3-5f0951e463ba.png)
![image](https://user-images.githubusercontent.com/3358282/87924236-81bfbe00-ca76-11ea-8689-fb20e37814ba.png)


It gets rid of the braces so the line count goes down, it also helps it keep similar looking to the getter, you can find the convention [here](https://kotlinlang.org/docs/reference/coding-conventions.html#expression-body-formatting).
With all the powerful features of kotlin it can sometimes be too easy to write a very complex function

```
fun crazyComplexStuff(someObj: SomeObj)=
with(someObje){
 this?.aProperty.let{
    it.map{ //blah}
     .anotherOperator()
     .anotherOperator()
}
} ?: someObj.defaultProperty. // more weird stuff
?: A_CONSTANT
```

and  abusing the scope function to get it to a 1 line return hampers the readability of the expression, always ask yourself does this read better.

The setters are a bit of a weird point because they have a return type of Unit (which all functions without a declared return type have), so even though you aren't actually returning anything the way kotlin works is that you implicitly return a Unit which states `a unit of work happened`

In this scenario I am fine with either really, I don't think either style has an explicit advantage in this file